src/cloudai/systems/slurm/slurm_system.py

```diff
@@ -713,7 +771,7 @@ def parse_nodes(self, nodes: List[str]) -> List[str]:
                 if len(parts) != 3:
                     raise ValueError("Format should be partition:group:num_nodes")
                 partition_name, group_name, num_nodes_str = parts
-                num_nodes = int(num_nodes_str)
+                num_nodes = int(num_nodes_str) if num_nodes_str != "max_avail" else num_nodes_str
```

Comment:
I don't think it's intuitive for num_nodes to have string values to support this feature. I believe it would be better design to expect an integer in num_nodes. Therefore, it might be preferable to add a new method to retrieve the number of idle nodes from a specified group and use it when num_nodes_str is set to `max_avail`. Please share your thoughts, @amaslenn & @srivatsankrishnan

Reply:
I understand, but if I do another method in the case num_nodes = 'max_avail', the method will be almost the same except for a few lines to differentiate the two cases.
What do you think @TaekyungHeo ?

Reply:
@ybenvidia , I have shared this with @srinivas212 last week.

Reply:
@TaekyungHeo With the design you want me to do, I would call group_nodes_by_state() twice: once in parse_nodes() to get the count of available nodes and once in get_available_nodes_from_group to actually allocate the nodes.
Maybe I only have to change the word num_nodes to be more generic and not only integers ?

Reply:
@ybenvidia, I'm not asking you to change the design. I just want to discuss this item with the larger group, that's all. It seems others aren't responding, so you should bring this up in the group as Jeff did.

Reply:
@TaekyungHeo Sure

Reply:
@amaslenn Maybe you have any idea about this comment ?

Reply:
I agree with Taekyung on this, but the only improvement I can suggest is naming: use `num_nodes_spec: str` instead of `num_nodes_str: str`.